### ISSUE TITLE
StructureGrowEvent doesn't fire when bonemeal is applied via Block.applyBoneMeal()

### ISSUE THREAD

**Issue by Frostauk** [PaperSpigot/Paper#6860]
### Expected behavior

When setting a block to a sapling and applying enough bonemeal to make it grow, I expected to see StructureGrowEvent fire for the tree as it would if grown by hand with bonemeal.

### Observed/Actual behavior

When applying bonemeal in this way, StructureGrowEvent does not fire.

### Steps/models to reproduce

Here is an example of how the bonemeal is applied to the sapling:
```
public static void doBrokenTreeGrowth(Location location) {
    location.getBlock().setType(Material.OAK_SAPLING);
    for (int i = 0; i < 15; i++) {
        location.getBlock().applyBoneMeal(BlockFace.DOWN);
    }
}
```

I have compiled a test plugin which will cause a tree to be grown using the method above upon left clicking the air. Whenever a structure is grown, a message containing how many times StructureGrowEvent was called versus how many time left-clicked tree were grown.

We should see the structures grown counter increase with each left click, but it does not.

Full test plugin code:
```
public class BugTesterMain extends JavaPlugin implements Listener {
	public static int brokenTreesGrown = 0;
	public static int structuresGrown = 0;

	@Override
	public void onEnable() {
		this.getServer().getPluginManager().registerEvents(this, this);
	}

	@EventHandler
	public void OnStructureGrow(StructureGrowEvent event) {
		structuresGrown++;
		event.getWorld().sendMessage(Component.text("Bonemeal Trees (left click) Grown: " + brokenTreesGrown));
		event.getWorld().sendMessage(Component.text("Structures Grown: " + structuresGrown));
	}

	@EventHandler
	public void OnPlayerInteract(PlayerInteractEvent event) {
		if (event.getHand() == EquipmentSlot.HAND && event.getAction() == Action.LEFT_CLICK_AIR) {
			doBrokenTreeGrowth(event.getPlayer().getLocation());

			brokenTreesGrown++;
			event.getPlayer().getWorld()
					.sendMessage(Component.text("Bonemeal Trees (left click) Grown: " + brokenTreesGrown));
			event.getPlayer().getWorld().sendMessage(Component.text("Structures Grown: " + structuresGrown));
		}
	}

	public static void doBrokenTreeGrowth(Location location) {
		location.getBlock().setType(Material.OAK_SAPLING);
		for (int i = 0; i < 15; i++) {
			location.getBlock().applyBoneMeal(BlockFace.DOWN);
		}
	}
}
```

### Plugin and Datapack List

Plugins (1): BugTester
There are 2 data packs enabled: [vanilla (built-in)], [file/bukkit (world)]

### Paper version

This server is running Paper version git-Paper-359 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: bc43f40)
You are running the latest version

### Other

N/A

**Comment by Machine-Maker** [comment]
Related: https://hub.spigotmc.org/jira/browse/SPIGOT-6895

**Comment by Machine-Maker** [comment]
Fixed in 1358d1e9146ca80d0ff84a644c7796d20b39765c
